### DISCUSSION TITLE
Auto-subtract files from new file lists

### DISCUSSION THREAD

**Discussion by ljestep** [airdcpp/airdcpp-windows#126] [Ideas]
As a data ~~hoarder~~ collector, I would like to see some added functionality to allow me to more quickly and easily weed out files that I don't want to download.

The 'Subtract List' utility along with the default of 'My List' helps tremendously with this, but there are some things that I think would help flesh it out and make it even more useful.

First of all, automate this so it always does it.  Obviously not everyone will want this as the default, so a setting would be needed for turning this on and off.

Second, Instead of only having your own list as a the default, make this a list of lists.  I maintain my own lists of 'bad' files that, for one reason or another, I don't want.  Some are trojans/fakes. Some are poor quality.  Some are files that I know to be broken.  

Finally, anything on my skip list.  I know I don't want them, so why even show them?  I've lost track of how many times I've found interesting looking directories only to find out that the only thing left after removing what I already have are hundreds of Thumbs.db files.

Thank you for your consideration, and for all the work you have done on AirDC++!

**Comment by maksis** [comment]
I've added the following API hooks for removing files/directories from filelists while they are being loaded: https://airdcpp.docs.apiary.io/#reference/filelists/hooks

There's also a new hook for removing incoming search results: https://airdcpp.docs.apiary.io/#reference/searching/hooks

Does that look good to you? The new functionality is not available in the current version.

Opening a filelist with 900k files and 100k directories takes 37 seconds with both hooks enabled (loading time is 4 seconds without hooks)

**Reply by ljestep** [reply]
Thank you!  

I'm out of town for the next couple of weeks.  I'll start looking at it when I get back.

**Reply by maksis** [reply]
Have you tried it yet? The endpoints are available in the latest beta.
